### DISCUSSION TITLE
Simple script to read and forward incoming messages in a private telegram channel

### DISCUSSION THREAD

**Discussion by diliprk** [python-telegram-bot/python-telegram-bot#3701] [Q&A]
Hi,
 I am trying to create a simple python script that I can deploy in some serverless env like AWS lambda where I continously listen to telegram messages incoming in  a private telegram channel and forward the received message _instantly_  to my personal channel or to another telegram user.

I came across this article which uses an older version of `python-telegram-bot` to listen to messages and forward it.
https://stackoverflow.com/questions/74910268/how-can-i-make-a-telegram-bot-listen-to-messages-from-a-different-chat-with-pyth

But this no longer works in the newer version of the library. I am new to `python-telegram-bot` library and the wiki page is a bit overwhelming.  Can anyone please suggest some simple code to get started for this requirement?

Best Regards,
Dilip

**Comment by lemontree210** [comment]
Hey, thanks for reaching out.

I would recommend looking at "[Your First Bot](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Extensions---Your-first-Bot)" to get the understanding of how the new version works. Then you can use the simple [Echobot example](https://docs.python-telegram-bot.org/en/stable/examples.echobot.html), but instead of just echoing the message back to the user, use [`Message.forward()`](https://docs.python-telegram-bot.org/en/stable/telegram.message.html#telegram.Message.forward) to forward it.

**Reply by diliprk** [reply]
Thanks @lemontree210  I am assuming to do all this, the bot I created from `BotFather` needs to be added as an administrator to the private channel right?
But what If I am just a subscriber to a closed private channel and I can only read incoming messages and don't have admin access to add my bot as an admin to this closed private channel or copy messages from this private channel? What is the solution in that case to read this incoming message in private closed channel and then forward it to my own channel?

**Reply by lemontree210** [reply]
> Thanks @lemontree210 I am assuming to do all this, the bot I created from `BotFather` needs to be added as an administrator to the private channel right? But what If I am just a subscriber to a closed private channel and I can only read incoming messages and don't have admin access to add my bot as an admin to this closed private channel or copy messages from this private channel? 

Hmm, I don't think the bot must be an administrator for this action (there's nothing about it in [Bot API](https://core.telegram.org/bots/api#forwardmessage))

**Comment by diliprk** [comment]
You can't add your bot to a private channel where you only have read only
access.

I only see paid options to do this https://docs.junction.space/forwarding/
but no tutorial or documentation anywhere.
Same situation here:
https://stackoverflow.com/questions/67343166/how-to-allow-telegram-bot-to-read-messages-in-a-private-channel-that-im-in-not

**Reply by lemontree210** [reply]
I think that that's how it is, yes. The bot doesn't need to be an admin but you do need to be to add it :)

**Reply by Poolitzer** [reply]
No the bot very much needs to be an admin to be in a channel

**Comment by kuhytitusdlz** [comment]
you can use https://github.com/iyear/tdl to read/export protected channel message, but import not support in this tool

**Comment by Ayax-BTC** [comment]
Can someone teach me all this please
